dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/relationship/DefaultRelationshipService.java

```diff
@@ -0,0 +1,445 @@
+            TrackedEntityType type = new TrackedEntityType();
+            type.setUid( enrollment.getTrackedEntityType() );
+            tei.setTrackedEntityType( type );
+            tei.setUid( enrollment.getTrackedEntityInstance() );
+
+            // tei owns all attributes in trackedEntityAttributeValues while programs only present a subset of them
+            // the program attributes are the ones attached to the enrollment
+            tei.setTrackedEntityAttributeValues(
+                enrollment.getAttributes().stream().map( this::map ).collect( Collectors.toSet() ) );
+
+            result.setEntityInstance( tei );
+        }
+
+        if ( StringUtils.isNotEmpty( enrollment.getOrgUnit() ) )
+        {
+            OrganisationUnit orgUnit = new OrganisationUnit();
+            orgUnit.setUid( enrollment.getOrgUnit() );
+            orgUnit.setName( enrollment.getOrgUnitName() );
+            result.setOrganisationUnit( orgUnit );
+        }
+
+        result.setGeometry( enrollment.getGeometry() );
+        result.setCreated( DateUtils.parseDate( enrollment.getCreated() ) );
+        result.setCreatedAtClient( DateUtils.parseDate( enrollment.getCreatedAtClient() ) );
+        result.setLastUpdated( DateUtils.parseDate( enrollment.getLastUpdated() ) );
+        result.setLastUpdatedAtClient( DateUtils.parseDate( enrollment.getLastUpdatedAtClient() ) );
+
+        Program program = new Program();
+        program.setUid( enrollment.getProgram() );
+        result.setProgram( program );
+
+        result.setStatus( enrollment.getStatus().getProgramStatus() );
+        result.setEnrollmentDate( enrollment.getEnrollmentDate() );
+        result.setIncidentDate( enrollment.getIncidentDate() );
+        result.setFollowup( enrollment.getFollowup() );
+        result.setEndDate( enrollment.getCompletedDate() );
+        result.setCompletedBy( enrollment.getCompletedBy() );
+        result.setStoredBy( enrollment.getStoredBy() );
+        result.setCreatedByUserInfo( enrollment.getCreatedByUserInfo() );
+        result.setLastUpdatedByUserInfo( enrollment.getLastUpdatedByUserInfo() );
+        result.setDeleted( enrollment.isDeleted() );
+
+        result
+            .setProgramStageInstances( enrollment.getEvents().stream().map( this::map ).collect( Collectors.toSet() ) );
+        result.setComments( enrollment.getNotes().stream().map( this::map ).collect( Collectors.toList() ) );
+
+        return result;
+    }
+
+    private TrackedEntityComment map( org.hisp.dhis.dxf2.events.event.Note note )
+    {
+        TrackedEntityComment result = new TrackedEntityComment();
+        result.setUid( note.getNote() );
+        result.setCommentText( note.getValue() );
+        result.setCreator( note.getStoredBy() );
+        result.setCreated( DateUtils.parseDate( note.getStoredDate() ) );
+        result.setLastUpdated( note.getLastUpdated() );
+        if ( note.getLastUpdatedBy() != null )
```

Comment:
this is just my personal taste, but I'd add a line break before the condition.

Reply:
All the mapping code from dxf2 to dhis-api is throwaway code. Maybe I need to make this more clear in the PR description 😅 So I won't do any cosmetics on this one 💅🏻 

The goal is to transition away from dxf2 in digestible steps while not breaking behavior.